catalog/plugin/catalog-plugin-query-monitor-plugin/src/main/java/ddf/catalog/querymonitor/QueryMonitorPlugin.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.querymonitor;
+
+import java.util.HashMap;
+import java.util.UUID;
+
+import ddf.catalog.plugin.PostFederatedQueryPlugin;
+import ddf.catalog.plugin.PreFederatedQueryPlugin;
+
+/**
+ * A {@link QueryMonitorPlugin} monitors ActiveSearches by maintaining a {@link HashMap} known as the SearchTable with a key of
+ * {@link UUID} and a value of {@link ActiveSearch}. {@link QueryMonitorPlugin} extends both {@link PreFederatedQueryPlugin}
+ * and {@link PostFederatedQueryPlugin} in order to process each search for every source. The {@link PreFederatedQueryPlugin}
+ * should be implemented to add the {@link ActiveSearch} to the SearchTable and the {@link PostFederatedQueryPlugin} should
+ * be implemented to remove the {@link ActiveSearch} from the SearchTable.
+ */
+
+public interface QueryMonitorPlugin<T> extends PreFederatedQueryPlugin, PostFederatedQueryPlugin {
+
+    /**
+     * Returns a copy of the ActiveSearchTable HashMap
+     *
+     * @return HashMap<UUID,ActiveSearch> copy of the ActiveSearchTable HashMap
+     */
+
+    HashMap<UUID, ActiveSearch> getSearchTable();
+
+    /**
+     * Sets a boolean which determines whether or not searches will be removed from the ActiveSearchTable upon search
+     * completion. A value of true will remove searches from the table as the searches finish and a value of false
+     * will keep the searches in the table after they complete.
+     *
+     * @param b boolean value. A value of false will keep ActiveSearches in the ActiveSearchTable even after
+     *          they finish (are no longer active). True will remove the ActiveSearches from the table as
+     *          the searches complete (default behavior is false)
+     */
+
+    void setRemoveFromTable(boolean b);
```

Comment:
The use of the term table in this interface seems odd to me.  I am not sure why this would be considered a table object.  I would expect the previous method to be something like `getActiveSearches` and this method to be something like `setKeepSearchAfterFinished`.